KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/listeners/LiveStreamEntry.java

```diff
@@ -572,6 +589,118 @@ public void onLiveStreamTranscoderCreate(ILiveStreamTranscoder liveStreamTransco
 			((LiveStreamTranscoder) liveStreamTranscoder).addActionListener(liveStreamTranscoderActionListener);
 		}
 	}
+	
+	class PacketListener implements IMediaStreamLivePacketNotify {
+		
+		protected static final String KNOWN_PTS = "known_pts";
+		// Maximum age allowed for a PTS in comparison to the current one
+		protected static final int MAX_PTS_AGE = 1000 * 10;
+		
+		@Override
+		@SuppressWarnings("unchecked")
+		public void onLivePacket(IMediaStream stream, AMFPacket packet) {
+
+			final int DEFAULT_CODEC = -1;
+			if (FLVUtils.isVideoKeyFrame(packet)) {
+				WMSProperties properties = getConnectionProperties(stream);
+				if(properties == null)
+					return; // Transcoded stream
+				
+				String streamName = stream.getName();
+				String entryId = getEntryIdFromStreamName(streamName);
+				if(entryId == null){
+					logger.info("Stream [" + streamName + "] does not match entry id");
+					return;
+				}
+				
+				// Calculate PTS
+				long pts = packet.getAbsTimecode();
+				int codec = properties.getPropertyInt(LiveStreamEntry.CLIENT_PROPERTY_CODEC_TYPE, DEFAULT_CODEC);
+				if ((codec == IVHost.CODEC_VIDEO_H264 || codec == IVHost.CODEC_VIDEO_H265) && packet.getSize() >= 5)
+					pts += BufferUtils.byteArrayToLong(packet.getData(), 2, 3);
+				
+				// Update PTS metadata and compare to related streams PTSes
+				ILiveStreamManager liveManager = KalturaServer.getManager(ILiveStreamManager.class);
+				Map<String, List<Long>> defaultValue = new HashMap<String, List<Long>>();
+				Map<String, List<Long>> streamsPtses = (Map<String, List<Long>>)liveManager.getMetadata(entryId, KNOWN_PTS, defaultValue);
+				addPts(streamsPtses, streamName, pts);
+				compareStreams(streamsPtses, streamName, pts);
+				cleanOldPtses(streamsPtses, streamName, pts);
```

Comment:
I must say this is confusing - 
1. Compare streams also removes from pts list (i.e. it also "cleans")
2. CleanOldPtses checks a *single* stream but notifies that a pts was not seen on other streams (i.e. it also "compares", and reports a message unrelated to its comparison)

I understand why this works, it's just not that clear from the code.
Can you please ensure there's a clear separation of concerns so that its clear what each method does and what we're checking?